test/github-package.test.js

```diff
@@ -8,617 +8,836 @@ import {fileExists, getTempDir} from '../lib/helpers';
-      const repository = contextPool.getContext(workdirPath).getRepository();
+      it('creates a context for the project', function() {
+        assert.isTrue(contextPool.getContext(workdirPath).isPresent());
+      });
 
-      repository.destroy();
+      context('when the repository is destroyed', function() {
+        beforeEach(function() {
+          const repository = contextPool.getContext(workdirPath).getRepository();
+          repository.destroy();
+        });
 
-      assert.isTrue(githubPackage.getActiveRepository().isAbsent());
+        it('uses an absent repository', function() {
+          assert.isTrue(githubPackage.getActiveRepository().isAbsent());
+        });
+      });
     });
 
-    // Don't worry about this on Windows as it's not a common op
-    if (process.platform !== 'win32') {
-      it('handles symlinked project paths', async function() {
+    context('with a symlinked repository project', function() {
+      beforeEach(async function() {
+        if (process.platform === 'win32') {
+          this.skip();
+        }
         const workdirPath = await cloneRepository('three-files');
         const symlinkPath = (await fs.realpath(temp.mkdirSync())) + '-symlink';
         fs.symlinkSync(workdirPath, symlinkPath);
         project.setPaths([symlinkPath]);
         await workspace.open(path.join(symlinkPath, 'a.txt'));
 
         await githubPackage.scheduleActiveContextUpdate();
+      });
+
+      it('uses a repository', async function() {
         await assert.async.isOk(githubPackage.getActiveRepository());
       });
-    }
+    });
   });
 
   describe('when there is a change in the repository', function() {
+    let atomEnv, githubPackage;
+    let workspace, project, commands, notificationManager;
+    let tooltips, deserializers, config, keymaps, styles;
+    let grammars, confirm, configDirPath, getLoadSettings;
+    let renderFn, contextPool, currentWindow;
+    let useLegacyPanels;
+
+    // Build Atom Environment and create the GitHub Package
+    beforeEach(async function() {
+      atomEnv = global.buildAtomEnvironment();
+      await disableFilesystemWatchers(atomEnv);
+
+      workspace = atomEnv.workspace;
+      project = atomEnv.project;
+      commands = atomEnv.commands;
+      deserializers = atomEnv.deserializers;
+      notificationManager = atomEnv.notifications;
+      tooltips = atomEnv.tooltips;
+      config = atomEnv.config;
+      keymaps = atomEnv.keymaps;
+      confirm = atomEnv.confirm.bind(atomEnv);
+      styles = atomEnv.styles;
+      grammars = atomEnv.grammars;
+      getLoadSettings = atomEnv.getLoadSettings.bind(atomEnv);
+      currentWindow = atomEnv.getCurrentWindow();
+      configDirPath = path.join(__dirname, 'fixtures', 'atomenv-config');
+      renderFn = sinon.stub().callsFake((component, element, callback) => {
+        if (callback) {
+          process.nextTick(callback);
+        }
+      });
+
+      useLegacyPanels = !workspace.getLeftDock;
```

Comment:
☝️ This check is for pretty ancient versions of Atom. We could probably get rid of it at this point.

Reply:
will remove